executor/src/witgen/global_constraints.rs

```diff
@@ -142,12 +138,13 @@ pub fn set_global_constraints<'a, T: FieldElement>(
 
     let mut retained_identities = vec![];
     let mut removed_identities = vec![];
-    for identity in identities.into_iter() {
+    for identity in fixed_data.identities.clone().into_iter() {
```

Comment:
Not a big fan of the clone...

Reply:
Hmm, I was able to avoid it by passing the IDs instead...